src/db.ts

```diff
@@ -1,23 +1,77 @@
 // db.ts
 import Dexie, { Table } from 'dexie'
 
-export interface Friend {
+export interface Student {
   id?: number
-  name: string
+  first_name: string
+  last_name: string
+  gender: 'male' | 'female' | 'nonbinary' | 'n/a'
   age: number
 }
 
-export class MySubClassedDexie extends Dexie {
+export interface FictionalGrade {
+  id?: number
+  student_id: number
+  v: boolean
+  kd: boolean
+  ca: boolean
+  i: boolean
+  e: boolean
+  l: boolean
+  go: boolean
+  mi: boolean
+}
+
+export interface InformationalGrade {
+  id?: number
+  student_id: number
+  v: boolean
+  kd: boolean
+  ca: boolean
+  i: boolean
+  e: boolean
+  l: boolean
+  go: boolean
+  mi: boolean
+}
+
+export interface SpellingGrade {
+  id?: number
+  student_id: number
+  phonetic_short_vowels: string
+  phonetic_consonant_blends: string
+  phonetic_consonant_digraphs: string
+  transitional_long_vowels: string
+  transitional_complex_vowels: string
+  fluent_inflectional_endings: string
+  fluent_multisyllabic_words_2_syllabes: string
+  advanced_multisyllabic_words_3_syllabes: string
+}
+
+export interface ReadingLevelGrade {
+  student_id: number
+  reading_level: number
+}
+
+export class StudentGradesDB extends Dexie {
   // 'friends' is added by dexie when declaring the stores()
   // We just tell the typing system this is the case
-  friends!: Table<Friend>
+  students!: Table<Student, number>
+  fictional_grades!: Table<FictionalGrade, number>
+  informational_grades!: Table<InformationalGrade, number>
+  spelling_grades!: Table<InformationalGrade, number>
+  reading_grades!: Table<InformationalGrade, number>
```

Comment:
From my understanding, this looks like a convenient little summary of all the students grades for different things like reading, spelling, etc. But shouldn't we include a date column as well? Otherwise, I'd be concerned that we wouldn't as easily be able to tell if we have an issue updating the table, and teachers might also want to see how grades change over time.

Is this table necessary? Or would it be better to query all of these grades dynamically from our other tables just so there's less to keep track of?

Reply:
Answered in [pull/1](https://github.com/phil208-ethics-project/phil208-ethics-project.github.io/pull/1)